### ISSUE TITLE
Media sources not working - Error 500 [2.4.2]

### ISSUE THREAD

**Issue by Dutch77** [magento/magento2#32341]
### Preconditions (*)

1. Magento version 2.4.2
2. php 7.4
3. nginx 1.19
4. MariaDB 10.4.17
5. elasticsearch 7.9.3

### Steps to reproduce (*)

1. Install fresh 2.4.2 project
2. switch it to production mode
3. Successfully upload image to any product
4. Try to visit image public URL
5. Get status code 500

### Expected result (*)

1. Getting 200 with image result :)

### Actual result (*)

```
NOTICE: PHP message: PHP Fatal error:  Uncaught RuntimeException: ObjectManager isn't initialized in /var/www/web/vendor/magento/framework/App/ObjectManager.php:36
Stack trace:
#0 /var/www/web/vendor/magento/framework/File/Mime.php(84): Magento\Framework\App\ObjectManager::getInstance()
#1 /var/www/web/pub/get.php(61): Magento\Framework\File\Mime->__construct()
#2 {main}
  thrown in /var/www/web/vendor/magento/framework/App/ObjectManager.php on line 36
127.0.0.1 -  02/Mar/2021:15:47:12 +0000 "GET /get.php" 500
```

### What exactly is wrong here
- It's because **2.4.2** version version of `Magento\Framework\File\Mime`. Class has now constructor that needs `Magento\Framework\Filesystem`.  https://github.com/magento/magento2/blob/2.4.2/lib/internal/Magento/Framework/File/Mime.php#L82
- If that is not provided, it tries to get that instance from `ObjectManager`
- but `ObjectManager` is instantiated later in flow see **line 76** **and line 64** https://github.com/magento/magento2/blob/2.4.2/pub/get.php

### Workaround
1. Move or duplicate **line 76** anywhere before **line 64** https://github.com/magento/magento2/blob/2.4.2/pub/get.php (not sure if that's not a performance issue)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hello @Dutch77 

Thank you for your report.

Can you please clarify step number 4 _Try to visit image public URL_ and if possible provide a screen or video.
Because for now, it's not clear.

Thank you in advance

**Comment by Dutch77** [comment]
> Hello @Dutch77
> 
> Thank you for your report.
> 
> Can you please clarify step number 4 _Try to visit image public URL_ and if possible provide a screen or video.
> Because for now, it's not clear.
> 
> Thank you in advance

I will clarify that
1. open some product in administration
2. upload an product image
4. save product
5. image is missing, browser placeholder instead see: https://ibb.co/3Fn8StC
6. this is the url of image `/media/catalog/product/r/e/red.png`
7. when visited, error 500 happen

**Comment by tangar76** [comment]
I think it has something to do with server settings.

When media image exists then server has to show it immediately. But for some reason handling this request is given to Magento `pub/get.php`.

I'm not good with server settings. It is quite hard for me to understand such behavior. But form PHP side I have found quick solution. Move Magento bootstrat initialization earlier in code. ObjectManager in initialized in `Bootstrap::create` method.

```diff
--- /pub/get.php
+++ /pub/get.php (Unsaved)
@@ -33,6 +33,15 @@
     )
 );
 $relativePath = $request->getPathInfo();
+
+// Materialize file in application
+$params = $_SERVER;
+if (empty($mediaDirectory)) {
+    $params[ObjectManagerFactory::INIT_PARAM_DEPLOYMENT_CONFIG] = [];
+    $params[Factory::PARAM_CACHE_FORCED_OPTIONS] = ['frontend_options' => ['disable_save' => true]];
+}
+$bootstrap = \Magento\Framework\App\Bootstrap::create(BP, $params);
+
 if (file_exists($configCacheFile) && is_readable($configCacheFile)) {
     $config = json_decode(file_get_contents($configCacheFile), true);
 
@@ -67,13 +76,6 @@
     }
 }
 
-// Materialize file in application
-$params = $_SERVER;
-if (empty($mediaDirectory)) {
-    $params[ObjectManagerFactory::INIT_PARAM_DEPLOYMENT_CONFIG] = [];
-    $params[Factory::PARAM_CACHE_FORCED_OPTIONS] = ['frontend_options' => ['disable_save' => true]];
-}
-$bootstrap = \Magento\Framework\App\Bootstrap::create(BP, $params);
 /** @var \Magento\MediaStorage\App\Media $app */
 $app = $bootstrap->createApplication(
     \Magento\MediaStorage\App\Media::class,

```

**Comment by Dutch77** [comment]
That is the exact workaround we're currently using :D

**Comment by engcom-Lima** [comment]
Hi @Dutch77 

I saw your issue, it is looking configuration settings related. Can you please provide me some screenshots and more steps to investigate this in details.

**Comment by engcom-Lima** [comment]
Hi @Dutch77 Can't reproduce the issue. I tested Magento ver. dev-2.4-develop. Hence closing this ticket.

Please find the attached Screenshots.
<img width="1292" alt="Screenshot 2021-06-16 at 6 21 50 PM" src="https://user-images.githubusercontent.com/60198022/122222509-f6da8400-cecf-11eb-9a31-6fa9258183fc.png">
<img width="1440" alt="Screenshot 2021-06-16 at 6 21 59 PM" src="https://user-images.githubusercontent.com/60198022/122222593-09ed5400-ced0-11eb-86d1-6e4a970c6c43.png">
<img width="1440" alt="Screenshot 2021-06-16 at 6 22 16 PM" src="https://user-images.githubusercontent.com/60198022/122222625-0fe33500-ced0-11eb-9a6a-dfce90607f51.png">

**Comment by vshatylo** [comment]
Hello, 
problem still reproduces on magento 2.4.3
it's hard to reproduce with one user or could be simulated with adding code "sleep(1)" into pub/get.php  
error occurs when many visitor surfs website.
Steps to reproduce
- upload new image for product or could be existing image 
- check media directory pub/media/catalog/product/cache/<image_path>, if file exists - remove it
- open cached product image with 10+ concurrent requests

**Comment by alexmtch** [comment]
Hello @engcom-Lima Make sure you goes trough this condition with you test case.
`if (file_exists($configCacheFile) && is_readable($configCacheFile))`

I've reproduced in local with this case.

**Comment by LB-Team-igniti** [comment]
> Hello, problem still reproduces on magento 2.4.3 it's hard to reproduce with one user or could be simulated with adding code "sleep(1)" into pub/get.php error occurs when many visitor surfs website. Steps to reproduce
> 
>     * upload new image for product or could be existing image
> 
>     * check media directory pub/media/catalog/product/cache/<image_path>, if file exists - remove it
> 
>     * open cached product image with 10+ concurrent requests

This is exactly how I could reproduce/enforce the issue locally as well. We see this error message on our LIVE system.

**Comment by LB-Team-igniti** [comment]
I just realized that this issue has been solved already in the latest 2.4-develop branch of magento and in version 2.4.5: https://github.com/magento/magento2/commit/f142625c0a04df4aab2fad73dc73d129e9e8fa8f
